### ISSUE TITLE
9 tests failing

### ISSUE THREAD

**Issue by harlantwood** [NejRemeslnici/tailwind-sorter#2]
Thanks for open sourcing this.  FYI tests are failing:

```sh
$ bundle install && bundle exec rspec

Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
Using bundler 2.2.23
Using diff-lcs 1.4.4
Using rspec-support 3.10.2
Using rspec-core 3.10.1
Using rspec-expectations 3.10.1
Using rspec-mocks 3.10.2
Using rspec 3.10.0
Bundle complete! 1 Gemfile dependency, 7 gems now installed.
Use `bundle info [gemname]` to see where a bundled gem is installed.
Calling `DidYouMean::SPELL_CHECKERS.merge!(error_name => spell_checker)' has been deprecated. Please call `DidYouMean.correct_error(error_name, spell_checker)' instead.
<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
F<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
F<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
F<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
F<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
F<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
F<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
.<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
.<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
.<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
F<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
F<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
.<internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require': cannot load such file -- tailwind_sorter/sorter (LoadError)
        from <internal:.rbenv/versions/3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
        from exe/tailwind_sorter:3:in `<main>'
F

Failures:

  1) tailwind_sorter when rewriting Slim input file with regexp 'slim_html' does basic class reordering
     Failure/Error: expect(run_tailwind_sorter(".rounded.my-4.block")).to eq(".block.my-4.rounded")
     
       expected: ".block.my-4.rounded"
            got: ".rounded.my-4.block"
     
       (compared using ==)
     # code/tailwind-sorter/spec/bin/tailwind_sorter_spec.rb:31:in `block (4 levels) in <top (required)>'

  2) tailwind_sorter when rewriting Slim input file with regexp 'slim_html' orders unknown classes first
     Failure/Error: expect(run_tailwind_sorter(".rounded.my-4.block.nr-class")).to eq(".nr-class.block.my-4.rounded")
     
       expected: ".nr-class.block.my-4.rounded"
            got: ".rounded.my-4.block.nr-class"
     
       (compared using ==)
     # code/tailwind-sorter/spec/bin/tailwind_sorter_spec.rb:35:in `block (4 levels) in <top (required)>'

  3) tailwind_sorter when rewriting Slim input file with regexp 'slim_html' orders variants to the end of the same group
     Failure/Error: expect(run_tailwind_sorter(".sm:block.block.lg:my-4.my-8")).to eq(".block.sm:block.my-8.lg:my-4")
     
       expected: ".block.sm:block.my-8.lg:my-4"
            got: ".sm:block.block.lg:my-4.my-8"
     
       (compared using ==)
     # code/tailwind-sorter/spec/bin/tailwind_sorter_spec.rb:39:in `block (4 levels) in <top (required)>'

  4) tailwind_sorter when rewriting Slim input file with regexp 'slim_html' orders multiple variants
     Failure/Error: expect(run_tailwind_sorter(".focus:hover:sm:block.my-4")).to eq(".sm:hover:focus:block.my-4")
     
       expected: ".sm:hover:focus:block.my-4"
            got: ".focus:hover:sm:block.my-4"
     
       (compared using ==)
     # code/tailwind-sorter/spec/bin/tailwind_sorter_spec.rb:43:in `block (4 levels) in <top (required)>'

  5) tailwind_sorter when rewriting Slim input file with regexp 'slim_html' removes duplicate classes
     Failure/Error: expect(run_tailwind_sorter(".block.my-4.block")).to eq(".block.my-4")
     
       expected: ".block.my-4"
            got: ".block.my-4.block"
     
       (compared using ==)
     # code/tailwind-sorter/spec/bin/tailwind_sorter_spec.rb:47:in `block (4 levels) in <top (required)>'

  6) tailwind_sorter when rewriting Slim input file with regexp 'slim_html' reorders classes up to equal sign (inline ruby)
     Failure/Error: expect(run_tailwind_sorter(".my-4.block = link_to(...)")).to eq(".block.my-4 = link_to(...)")
     
       expected: ".block.my-4 = link_to(...)"
            got: ".my-4.block = link_to(...)"
     
       (compared using ==)
     # code/tailwind-sorter/spec/bin/tailwind_sorter_spec.rb:51:in `block (4 levels) in <top (required)>'

  7) tailwind_sorter when rewriting Slim input file with regexp 'ruby_class_attribute' does basic class reordering
     Failure/Error: expect(run_tailwind_sorter('class: "rounded my-4 block"')).to eq('class: "block my-4 rounded"')
     
       expected: "class: \"block my-4 rounded\""
            got: "class: \"rounded my-4 block\""
     
       (compared using ==)
     # code/tailwind-sorter/spec/bin/tailwind_sorter_spec.rb:69:in `block (4 levels) in <top (required)>'

  8) tailwind_sorter when rewriting Slim input file with regexp 'ruby_class_attribute' reorders classes only before interpolation
     Failure/Error:
       expect(run_tailwind_sorter('class: "rounded my-4 block #{...} mx-8"')).to(
         eq('class: "block my-4 rounded #{...} mx-8"')
       )
     
       expected: "class: \"block my-4 rounded \#{...} mx-8\""
            got: "class: \"rounded my-4 block \#{...} mx-8\""
     
       (compared using ==)
     # code/tailwind-sorter/spec/bin/tailwind_sorter_spec.rb:73:in `block (4 levels) in <top (required)>'

  9) tailwind_sorter when only checking the classes prints an error when css classes order is not proper
     Failure/Error:
       expect { run_tailwind_sorter_in_warning_mode(".rounded.my-4.block") }.to(
         output(/^#{test_file}:1:CSS classes are not sorted well/).to_stdout_from_any_process
       )
     
       expected block to output /^tmp\/tailwind_sorter_test.slim:1:CSS classes are not sorted well/ to stdout, but output nothing
       Diff:
       @@ -1 +1 @@
       -/^tmp\/tailwind_sorter_test.slim:1:CSS classes are not sorted well/
       +""
       
     # code/tailwind-sorter/spec/bin/tailwind_sorter_spec.rb:96:in `block (3 levels) in <top (required)>'

Finished in 0.81946 seconds (files took 0.04492 seconds to load)
13 examples, 9 failures

Failed examples:

rspec ./spec/bin/tailwind_sorter_spec.rb:30 # tailwind_sorter when rewriting Slim input file with regexp 'slim_html' does basic class reordering
rspec ./spec/bin/tailwind_sorter_spec.rb:34 # tailwind_sorter when rewriting Slim input file with regexp 'slim_html' orders unknown classes first
rspec ./spec/bin/tailwind_sorter_spec.rb:38 # tailwind_sorter when rewriting Slim input file with regexp 'slim_html' orders variants to the end of the same group
rspec ./spec/bin/tailwind_sorter_spec.rb:42 # tailwind_sorter when rewriting Slim input file with regexp 'slim_html' orders multiple variants
rspec ./spec/bin/tailwind_sorter_spec.rb:46 # tailwind_sorter when rewriting Slim input file with regexp 'slim_html' removes duplicate classes
rspec ./spec/bin/tailwind_sorter_spec.rb:50 # tailwind_sorter when rewriting Slim input file with regexp 'slim_html' reorders classes up to equal sign (inline ruby)
rspec ./spec/bin/tailwind_sorter_spec.rb:68 # tailwind_sorter when rewriting Slim input file with regexp 'ruby_class_attribute' does basic class reordering
rspec ./spec/bin/tailwind_sorter_spec.rb:72 # tailwind_sorter when rewriting Slim input file with regexp 'ruby_class_attribute' reorders classes only before interpolation
rspec ./spec/bin/tailwind_sorter_spec.rb:95 # tailwind_sorter when only checking the classes prints an error when css classes order is not proper

```

**Comment by harlantwood** [comment]
I rolled back one commit at a time until tests passed, most recent passing commit is this one:

```
* 7c2f126 - (HEAD -> dev, tag: original-script) fix-typo (#1) (5 months ago) <Juzer Shakir>
```

**Comment by borama** [comment]
Hey, thanks for the report and sorry for the delay, I completely missed notifications in this repo! :sweat_smile: 

I managed to reproduce this on ruby 3.1.0 (the tests pass on 3.3.0). I will look into this deeper, thanks.

**Comment by borama** [comment]
In the end, this was not about ruby versions but just about an issue with the `LOAD_PATH` when running tests. I had the gem locally installed in my ruby, which circumvented the `LOAD_PATH` issue too, so that's why the tests succeeded in my case. [Fixed](https://github.com/NejRemeslnici/tailwind-sorter/commit/5e7b4fa7ea0038e5197ab890ea0e836e0d7081c4) in main and it will be released in the gem version 0.4, soon. Thanks!
